sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -215,7 +215,7 @@ namespace Azure { namespace Core { namespace Http {
     std::map<std::string, std::string> m_retryHeaders;
     std::map<std::string, std::string> m_retryQueryParameters;
     // Work only with streams
-    std::unique_ptr<BodyStream> m_bodyStream;
+    BodyStream& m_bodyStream;
```

Comment:
Why is this changed to reference again? It was just changed to unique_ptr several days ago.

I vaguely remember that we don't allow reference in member variables.

Reply:
here https://azure.github.io/azure-sdk/cpp_design.html#const-and-reference-members

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/238

Reply:
It would be best for us to avoid this kind of breaking changes in the future. Changing from A to B is fine, but changing from A to B to C then back to A seems to be badly planned or designed.
Also, there is no detailed information in the linked issue for the Storage team to know the reasoning behind this change.

Reply:
We're experimenting with different designs and getting input from Billy/Casey. Right now, I cannot guarantee that this will not change again - in fact, we're meeting on it again today. We want to get this right and there are nuances. C++ makes it very difficult; in other languages this was trivial. Please just bear with us a while longer.